pkg/services/accesscontrol/database/database.go

```diff
@@ -277,7 +278,12 @@ func (s *AccessControlStore) SearchUsersPermissions(ctx context.Context, orgID i
 }
 
 // GetUsersBasicRoles returns the list of user basic roles (Admin, Editor, Viewer, Grafana Admin) indexed by UserID
-func (s *AccessControlStore) GetUsersBasicRoles(ctx context.Context, userFilter []int64, orgID int64) (map[int64][]string, error) {
+// Special case for anonymous users, they don't have a user ID, we return the basic roles for them
+func (s *AccessControlStore) GetUsersBasicRoles(ctx context.Context, userIDFilter []int64, orgID int64) (map[int64][]string, error) {
+	// Special case for anonymous users, they don't have a user ID
+	if len(userIDFilter) == 1 && userIDFilter[0] == 0 {
```

Comment:
This function is a part of `database` package which is responsible to return data that located in database. So this special case feels a bit out of responsibility of this package. Maybe it makes sense to move it upper (this function called from `searchUserPermissions` in `pkg/services/accesscontrol/acimpl/service.go`). I think business logic should be located there.

Reply:
+1 to that, I also think we should handle this higher up.

Reply:
moved the logic up. continuing the conversation on what we should look for on the other review comment